src/Microsoft.DotNet.Build.Tasks/PrereleaseResolveNuGetPackageAssets.cs

```diff
@@ -902,7 +902,7 @@ private string GetNuGetPackagePath(string packageId, string packageVersion)
                 else
                 {
                     // Try a lower-case path.
-                    string lowerCasePath = packagePath.ToLowerInvariant();
+                    string lowerCasePath = Path.Combine(packagesFolder, packageId.ToLowerInvariant(), packageVersion);
```

Comment:
Can you double check if NuGet is also ToLower'ing the packageVersion? I believe that can also contain uppercase characters.